loguru_cmake_example/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+cmake_minimum_required(VERSION 3.10)
+
+project(loguru_cmake_example CXX)
+
+# Choose any ONE of the following three methods to import loguru into your cmake
+#
+# How do I set loguru compile-time flags?!
+#
+#   This example file demonstrates setting the 'LOGURU_WITH_STREAMS' option
+#
+#   For method 1 and 2:
+#     loguru is compiled alongside your project so all loguru compile-time flags
+#     can be set in your project as cmake variables!
+#
+#   For method 3:
+#     This method uses a pre-compiled copy of loguru, so the compile-time flags
+#     cannot be changed by your project. They must be set when compiling loguru.
+
+# ------------------------------------------------------------------------------
+# --- Method 1: Add loguru as a sub-directory (great with git submodules!)
+# ------------------------------------------------------------------------------
+
+# set any loguru compile-time flags before adding the subdirectory
+set(LOGURU_WITH_STREAMS TRUE)
+add_subdirectory(loguru) # defines target 'loguru::loguru'
+
+# ------------------------------------------------------------------------------
+# --- Method 2: Fetch from an external repository
+# ------------------------------------------------------------------------------
+
+# NOTE: The FetchContent functions shown here were introduced in CMake 3.14
+
+include(FetchContent)
```

Comment:
Are we not mixing three versions in the same file now? Or how do we select which method to use?

Reply:
Or is this meant as a template?

Reply:
Sorry, thought I was in the other CMakeLists.txt file.

Reply:
I can see how this is a bit unclear.
I created that `CMakeLists.txt` file just to demonstrate the syntax for each import method. That CMakeLists.txt file is not meant to be runnable as-is.